magnum_cluster_api/resources.py

```diff
@@ -271,6 +271,88 @@ def get_object(self) -> objects.ClusterResourceSet:
         )
 
 
+class StorageClassesConfigMap(ClusterBase):
+    def __init__(
+        self,
+        context: context.RequestContext,
+        api: pykube.HTTPClient,
+        cluster: magnum_objects.Cluster,
+    ):
+        self.context = context
+        self.api = api
+        self.cluster = cluster
+
+    def get_object(self) -> pykube.ConfigMap:
+        osc = clients.get_openstack_api(self.context)
+        volume_types = osc.cinder().volume_types.list()
+        data = {}
+        for vt in volume_types:
+            data[f"{vt.name}.yaml"] = yaml.dump(
+                {
+                    "apiVersion": objects.StorageClass.version,
+                    "kind": objects.StorageClass.kind,
+                    "metadata": {
+                        "annotations": {
+                            "storageclass.kubernetes.io/is-default-class": "true"
+                        }
+                        if vt.name == "__DEFAULT__"
+                        else {},
+                        "name": vt.name,
+                    },
```

Comment:
hmm, it's possible that the default storage class could be called something else other than `__DEFAULT__`

https://docs.openstack.org/api-ref/block-storage/v3/?expanded=list-all-volume-types-detail,show-default-volume-type-detail#show-default-volume-type

it's advised not to use/rely on `__DEFAULT__` so use that instead.

Reply:
ok, this is the exact method
https://github.com/openstack/python-cinderclient/blob/master/cinderclient/v3/volume_types.py#L116-L121